### PR TITLE
Deal with a bunch of TODOs

### PR DESCRIPTION
**Pull request by DavidBoike**

### CODE REVIEW

File: src/ServiceControl.Persistence.RavenDb/RavenDbPersistence.cs [hunk 1]
```diff
@@ -78,21 +78,6 @@ public IPersistenceLifecycle CreateLifecycle()
...
 
-        // TODO: Make sure this stuff from PersistenceHostBuilderExtensions is accounted for here
-
-        //var documentStore = new EmbeddableDocumentStore();
-        //RavenBootstrapper.Configure(documentStore, settings);
-
-        //hostBuilder.ConfigureServices(serviceCollection =>
-        //{
-        //    serviceCollection.AddSingleton<IDocumentStore>(documentStore);
-        //    serviceCollection.AddHostedService<EmbeddedRavenDbHostedService>();
```
Comment (DavidBoike) [inline src/ServiceControl.Persistence.RavenDb/RavenDbPersistence.cs:89]: These two are covered by the RavenLifetime thing

File: src/ServiceControl.Persistence.RavenDb/RavenDbPersistence.cs [hunk 2]
```diff
@@ -78,21 +78,6 @@ public IPersistenceLifecycle CreateLifecycle()
...
-        //    serviceCollection.AddCustomCheck<CheckRavenDBIndexErrors>();
-        //    serviceCollection.AddCustomCheck<CheckRavenDBIndexLag>();
```
Comment (DavidBoike) [inline src/ServiceControl.Persistence.RavenDb/RavenDbPersistence.cs:91]: These two are registered up above in the same file

File: src/ServiceControl/Operations/ErrorProcessor.cs [hunk 3]
```diff
@@ -117,7 +117,7 @@ async Task ProcessMessage(MessageContext context, IIngestionUnitOfWork unitOfWor
 
                 var processingAttempt = failedMessageFactory.CreateProcessingAttempt(
                     context.Headers,
-                    new Dictionary<string, object>(metadata), // TODO: metadata is already a dictionary, it this really needed?
+                    new Dictionary<string, object>(metadata),
```
Comment (DavidBoike) [inline src/ServiceControl/Operations/ErrorProcessor.cs:120]: I think this is attempting to duplicate the dictionary so modifications don't affect the original on any sort of retry.

File: src/ServiceControl/Recoverability/API/FailureGroupsController.cs [hunk 4]
```diff
@@ -85,7 +85,7 @@ public async Task<HttpResponseMessage> GetAllGroups(string classifier = "Excepti
                 classifierFilter = null;
             }
 
-            var results = await groupFetcher.GetGroups(classifier, classifierFilter); // TODO: Analyze what to do with the GroupFetcher dependency
+            var results = await groupFetcher.GetGroups(classifier, classifierFilter);
```
Comment (DavidBoike) [inline src/ServiceControl/Recoverability/API/FailureGroupsController.cs:88]: It works, I don't think we need to analyze

File: src/ServiceControl.Persistence.RavenDb/FailedMessageViewIndexNotifications.cs [hunk 5]
```diff
@@ -9,7 +9,7 @@
     using Persistence;
     using Raven.Client;
 
-    class FailedMessageViewIndexNotifications  // TODO: Must be registered as single and as hosted service
+    class FailedMessageViewIndexNotifications
```
Comment (DavidBoike) [inline src/ServiceControl.Persistence.RavenDb/FailedMessageViewIndexNotifications.cs:12]: I checked, it is.

File: src/ServiceControl.Persistence.RavenDb/ErrorMessagesDataStore.cs [hunk 6]
```diff
@@ -692,10 +691,9 @@ public Task RemoveFailedMessageRetryDocument(string uniqueMessageId)
                 }
             }
 
-            return ids.ToArray(); // TODO: Currently returning array as all other API's return arrays and not IEnumerable<T>
+            return ids.ToArray();
         }
 
-        // TODO: How is this different than what RavenAttachmentBodyStorage.TryFetch is doing? Is this implemented twice?
```
Comment (DavidBoike) [inline src/ServiceControl.Persistence.RavenDb/ErrorMessagesDataStore.cs:698]: This was my misunderstanding of the 2 methods in Raven35 of storing bodies.

File: src/ServiceControl.Persistence.RavenDb/ErrorMessagesDataStore.cs [hunk 7]
```diff
@@ -343,8 +343,6 @@ SortInfo sortInfo
                         x => (object)x.Value
                         );
 
-                Guard.Assert(false, "TODO: Check how to convert dictionary item VALUES, currently return object which must be typed");
```
Comment (DavidBoike) [inline src/ServiceControl.Persistence.RavenDb/ErrorMessagesDataStore.cs:346]: I don't get what was going on here

**Review by tmasternak — APPROVED** [review]

File: src/ServiceControl.Persistence.RavenDb/ErrorMessagesDataStore.cs (hunk 7 above)
Comment (ramonsmits) [reply src/ServiceControl.Persistence.RavenDb/ErrorMessagesDataStore.cs:346]: Its obsolete

**Review by ramonsmits — APPROVED** [review]

**Pull request merged**